packages/blade/src/components/TextArea/_decisions/decisions.md

```diff
@@ -0,0 +1,66 @@
+| onClearButtonClicked | `() => {}`                       |         | Used in conjunction with showClearButton prop. If clicked it removes the contents of the TextArea. Users can also hook their own analytics code or do something else.                                                                                        |          |
+| maxLines             | `number`                         | `2`     | Sets the number of lines (or rows) of the text area, minimum is `2` maximum is `5`                                                                                                                                                                           |          |
+| maxCharacters        | `number`                         |         | Used to show a character counter under the input field                                                                                                                                                                                                       |          |
+| validationState      | `error , success`                |         | Applies the style to the input field based on validation status and shows either errorText or successText respectively                                                                                                                                       |          |
+| helpText             | `string`                         |         | Shown when we want to add some hint to the input field. Displayed under the input field. Only one of helpText, errorText or successText is shown at a time in the priority order as errorText, successText, helpText                                         |          |
+| errorText            | `string`                         |         | Shown when the validationState of the input field is set to Error.                                                                                                                                                                                           |          |
+| successText          | `string`                         |         | Shown when the validationState of the input field is set to Success                                                                                                                                                                                          |          |
+| autoFocus            | `boolean`                        | `false` | The autofocus global attribute is a Boolean attribute indicating that an element should be focused on page load. Web Reference, Native Reference                                                                                                             |          |
+
+
+## Open Questions: 
+
+- [ ] Should the `onClearButton` by default clear the TextArea contents? Or should users handle that logic?
```

Comment:
I think it should just clear the text, we shouldn't make that part configurable. But users should be able to hook in their callback to call additionally to the above.

Reply:
Okay so if users do this: 

<TextArea onClearButtonClicked={() => { alert('cleared') }} />

This will clear the textarea and then show the alert. 

This?

Reply:
Yes, that's how I would expect it to work.

Reply:
what I've done for textinput is
1. if it is uncontrolled then clear it
2. if it is controlled then do nothing and let the users clear it since they will be managing the state in their application and if we clear and they don't then there will be discrepancy in their state but visually the input will be empty